common/failure/failure.go

```diff
@@ -89,3 +92,43 @@ func TruncateWithDepth(f *failurepb.Failure, maxSize, maxDepth int) *failurepb.F
 
 	return newFailure
 }
+
+// IsRetryable determines if a failure is retryable based on its type and non-retryable types list.
+func IsRetryable(failure *failurepb.Failure, nonRetryableTypes []string) bool {
```

Comment:
FWIW the failure you get from the API must always be an ApplicationFailure instance.